src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -2895,8 +2907,10 @@ IsJoinClause(Node *clause)
 	List *argumentList = NIL;
 	Node *leftArgument = NULL;
 	Node *rightArgument = NULL;
-	List *leftColumnList = NIL;
-	List *rightColumnList = NIL;
+	Node *strippedLeftArgument = NULL;
+	Node *strippedRightArgument = NULL;
+	NodeTag leftArgumentType = T_Invalid;
```

Comment:
Minor: We probably don't need `leftArgumentType ` and `rightArgumentType `, which makes it slightly harder to follow. Can't we move `nodeTag(strippedLeftArgument)` and `nodeTag(strippedRightArgument)` to the if statements instead?

Reply:
replaced them with
```c
if (IsA(strippedLeftArgument, Var) && IsA(strippedRightArgument, Var))
```